src/components/Videos.vue

```diff
@@ -128,238 +129,35 @@ export default {
 			this.updateHeightWidth()
 		},
 
+		// Fetch caption tracks and build HTML5 block
+		fetchTracks() {
+			const url = generateUrl('/apps/viewer/video/tracks')
+			const params = { params: { videoPath: this.filename } }
+			axios.get(url, params).then(response => {
+				const davDir = this.davPath.replace(/[^/]*$/, '')
```

Comment:
What does this do?

Reply:
@skjnldsv, this computes the "dav" path to the folder where the movie and the caption tracks reside.
Then this path is concatenated with the caption tracks basename to produce "dav" paths as expected by the plyr library.

Or do you mean I should comment the code?
Or something else?

Reply:
No I mean, the replace part.

Reply:
OK. It is a regular expression for a pattern match.
/[^/]*$/
It means: remove the end of the path up to the first slash.
E.g.:
  - 'titi/toto.mkv' --> 'titi/'
  - 'titi/' --> 'titi/'
  - 'toto.mkv' --> ''

Reply:
We have dedicated methods for those iirc in `@nextcloud/paths`
https://www.npmjs.com/package/@nextcloud/paths